htdocs/adherents/class/adherent.class.php

```diff
@@ -966,30 +966,32 @@ public function delete($rowid, $user, $notrigger = 0)
 	 *    @param	int		$nosyncuser		Do not synchronize linked user
 	 *    @return   string           		If OK return clear password, 0 if no change, < 0 if error
 	 */
-	public function setPassword($user, $password = '', $isencrypted = 0, $notrigger = 0, $nosyncuser = 0)
+	public function setPassword($user, $password = '', $password_indatabase_crypted = '', $isencrypted = 0, $notrigger = 0, $nosyncuser = 0)
```

Comment:
When you add a new parameter, it must always be at end or you break compatibility.
You must add it into the header doc of method or Continuous integration tests will fails.

Also, i don't understand why you need to add a parameter to setPassword. setPassword goal is to change the password from a new value, so only the new value should be required ?

Reply:
create_from_member was taking from database an empty password. "member" did have pass_indatabase_crypted but wasnt being used by setPassword so I was trying to add it too and use pass_indatabase_crypted instead of password when possible.

I didn't think in break compatibility, sorry. Now I just add a parameter called "alreadyencrypted" at end of the method setPassword that check is the password added is encrypted or not. I hope this solution is better than the previos one. 

I change both setPassword (in user.class and adherent.class) to avoid adherent.class encryp again the password cause was making imposible to login via public formulary.